File: src/main/java/com/jcloisterzone/action/TowerPieceAction.java
package com.jcloisterzone.action;

import com.jcloisterzone.board.Position;
import com.jcloisterzone.game.capability.TowerCapability;
import com.jcloisterzone.game.capability.TunnelCapability.Tunnel;

import io.vavr.collection.Set;

public class TowerPieceAction extends SelectTileAction {

	private final TowerCapability.TowerToken token;
	
    public TowerPieceAction(Set<Position> options, TowerCapability.TowerToken token) {
        super(options);
        this.token = token;
    }

    public TowerCapability.TowerToken getToken() {
        return token;
    }

}

File: src/main/ts/com/jcloisterzone/action/TowerPieceAction.ts
import type { Set } from "../../../io/vavr/Set.js";
import type { Position } from "../board/Position.js";
import type { TowerCapability } from "../game/capability/TowerCapability.js";
import { SelectTileAction } from "./SelectTileAction.js";

/** Offer to place a tower piece (TOWER_PIECE / BLACK / WHITE) on a tower position. */
export class TowerPieceAction extends SelectTileAction {
  static readonly simpleName = "TowerPieceAction";

  constructor(
    options: Set<Position>,
    private readonly token: TowerCapability.TowerToken,
  ) {
    super(options);
  }

  getToken(): TowerCapability.TowerToken {
    return this.token;
  }
}
